fastly/block_fastly_service_v1_logging_digitalocean_test.go

```diff
@@ -0,0 +1,314 @@
+		Period:            3601,
+		Format:            "%h %l %u %t \"%r\" %>s %b %T",
+		TimestampFormat:   "%Y-%m-%dT%H:%M:%S.000",
+		GzipLevel:         2,
+		FormatVersion:     2,
+		MessageType:       "blank",
+		Placement:         "none",
+		ResponseCondition: "response_condition_test",
+	}
+
+	log2 := gofastly.DigitalOcean{
+		Version:           1,
+		Name:              "another-digitalocean-endpoint",
+		BucketName:        "bucket2",
+		AccessKey:         "access2",
+		SecretKey:         "secret2",
+		Domain:            "nyc3.digitaloceanspaces.com",
+		PublicKey:         pgpPublicKey(t),
+		Path:              "two/",
+		Period:            3600,
+		Format:            "%h %l %u %t \"%r\" %>s %b",
+		TimestampFormat:   "%Y-%m-%dT%H:%M:%S.000",
+		GzipLevel:         1,
+		FormatVersion:     2,
+		MessageType:       "classic",
+		Placement:         "none",
+		ResponseCondition: "response_condition_test",
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckServiceV1Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceV1DigitalOceanConfig(name, domain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
+					testAccCheckFastlyServiceV1DigitalOceanAttributes(&service, []*gofastly.DigitalOcean{&log1}),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "name", name),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "logging_digitalocean.#", "1"),
+				),
+			},
+
+			{
+				Config: testAccServiceV1DigitalOceanConfig_update(name, domain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
+					testAccCheckFastlyServiceV1DigitalOceanAttributes(&service, []*gofastly.DigitalOcean{&log1_after_update, &log2}),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "name", name),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "logging_digitalocean.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckFastlyServiceV1DigitalOceanAttributes(service *gofastly.ServiceDetail, digitalocean []*gofastly.DigitalOcean) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		conn := testAccProvider.Meta().(*FastlyClient).conn
+		digitaloceanList, err := conn.ListDigitalOceans(&gofastly.ListDigitalOceansInput{
+			Service: service.ID,
+			Version: service.ActiveVersion.Number,
+		})
+
+		if err != nil {
+			return fmt.Errorf("[ERR] Error looking up DigitalOcean Spaces Logging for (%s), version (%d): %s", service.Name, service.ActiveVersion.Number, err)
+		}
+
+		if len(digitaloceanList) != len(digitalocean) {
+			return fmt.Errorf("DigitalOcean Spaces List count mismatch, expected (%d), got (%d)", len(digitalocean), len(digitaloceanList))
+		}
+
+		log.Printf("[DEBUG] digitaloceanList = %#v\n", digitaloceanList)
```

Comment:
Is this still needed?

Reply:
This does still help with debugging global state.